awx/main/scheduler/dependency_graph.py

```diff
@@ -72,7 +72,12 @@ def project_update_blocked_by(self, job):
         return self.get_item(self.PROJECT_UPDATES, job.project_id)
 
     def inventory_update_blocked_by(self, job):
-        return self.get_item(self.INVENTORY_SOURCE_UPDATES, job.inventory_source_id)
+        inventory_source_block = self.get_item(self.INVENTORY_SOURCE_UPDATES, job.inventory_source_id)
+        project_block = self.get_item(self.PROJECT_UPDATES, job.inventory_source.source_project_id)
```

Comment:
You should check if `job.inventory_source` is null before referencing `job.inventory_source.source_project_id`, because at this point it's not clear that it's SCM type, or we could be in a deletion cascade or something.

I'm also unsure if this condition is a good idea. Your other logic already covers creating and linking a project update dependency. This looks like it would block the inventory update if another update started after its dependency... which I don't really want.

Reply:
the task manager links the project update as a dependency to the _job_, not the inventory update. Maybe this should be a dependency off of the inventory update itself